src/modules/openrccfg/main.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# === This file is part of Calamares - <http://github.com/calamares> ===
+#
+#   Copyright 2016, Artoo <artoo@manjaro.org>
+#   Copyright 2017, Philip Müller <philm@manjaro.org>
+#   Copyright 2018, Artoo <artoo@artixlinux.org>
+#
+#   Calamares is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   Calamares is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with Calamares. If not, see <http://www.gnu.org/licenses/>.
+
+import libcalamares
+
+from libcalamares.utils import target_env_call
+from os.path import exists, join
+
+
+class OpenrcController:
+    """This is the service controller
+    """
+
+    def __init__(self):
+        self.root = libcalamares.globalstorage.value('rootMountPoint')
+        self.services = libcalamares.job.configuration.get('services', [])
+
+    def update(self, state):
+        """Update init scripts
+        """
+
+        for svc in self.services[state]:
+            if exists(self.root + "/etc/init.d/" + svc["name"]):
```

Comment:
Should something be logged if it doesn't exist? That means there's a mismatch between the configuration, and the installed system.

Reply:
To be honest, I don't think this even needs to be in here. I assume distro maintainers will know exactly what services are present on the ISO image, so this check is kinda redundant.

Reply:
@V3n3RiX:
> I assume distro maintainers will know exactly what services are present on the ISO image

This is not really true. There can be custom remixes of the ISO based on the packages (including the Calamares configuration) from the upstream distribution. The upstream distribution (which writes the configuration) has no way to know what ends up on such a custom ISO.

Reply:
I don't think is upstream distributions job to even care about what ends up in such remixes. You have your own Fedora remix, do you take everything as is, or do you make adjustments to suit your needs?

Reply:
Well, the situation is a bit particular because I maintain both the remix (Kannolo) and the Fedora package for Calamares. I tried hard to make the configuration shipped by the Fedora package work for remixes too. (I even pick up the branding from where a Fedora Remix normally puts it to autogenerate a Calamares branding.) I still do a couple of tweaks to the Kannolo Calamares configs, but the idea is that Calamares ought to work even without those tweaks.

Reply:
Interesting @kkofler 

We take a slight different approach to the conf files.
upstream installs the standard conf files in /sur/share/calamares/modules, we override them in /etc/calamares/modules.
But, the build script for iso writes some of the distro overrides conf, eg services based on the build script's conf. Ie, we hand over the the conf values of the buildscript to calamares module conf files where required to get around the issue of packaging hell with basically variable conf.

Reply:
The exact way the conf files are handled in the Fedora Calamares package is that I install my configuration to `/usr/share/calamares/modules` (I patch the upstream conf files to the settings I need), and remixes such as Kannolo can then write further customizations to `/etc/calamares/modules` (which ships empty in the package).

Reply:
as for the 'hard-coding' issue, the solution may very simple:
```
SYNOPSIS
     rc-update [-s, --stack] add service [runlevel ...]
 # rc-update add cupsd
 * rc-update: cupsd already installed in runlevel `default'; skipping
# rc-update del cupsd
 * service cupsd removed from runlevel default
 # rc-update add cupsd
 * service cupsd added to runlevel default
```
There is no need to use 'runlevel' at all. Without it, rc-update will default .. to *default*.
And I think there is a serious reason to be designed like that ... but we have to ask the openrc dev guys for it ...